### ISSUE TITLE
useless_else_on_loop doesn't take into account enclosed return statement

### ISSUE THREAD

**Issue by ceridwen** [PyCQA/pylint#1592]
### Steps to reproduce

Run pylint on code that looks like:

```python
    for _ in range(times):
        time.sleep(sleep)
        try:
            value = function(*args, **kws)
        except exception as error:
            print(error)
        if condition(value):
            return value
    else:
         <code>
```

### Current behavior

pylint suggests that the else clause here is useless despite there being a code path that leads to the return.

### Expected behavior

No useless_else_on_loop warning.

### pylint --version output

__main__.py 1.7.2,
astroid 1.5.3
Python 3.6.1 (default, Apr  4 2017, 09:40:21)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

### Comments

A simple search inside the AST resulting from parsing the above code should be able to find if there are any break or return statements inside it.  astroid isn't flow-sensitive so it can't figure out if the return or break is reachable, but that's not needed to recognize this warning doesn't apply here.

**Comment by PCManticore** [comment]
I don't think I approve of this. In my reasoning, if one has a ``return`` in their loop, without having a ``break``, then the ``else`` clause is superfluos, since that branch is always taken in case the ``return`` doesn't get triggered. This is a bit different than the ``break`` case, which can be viewed rather as an ``if-else`` conditional, in the sense that ``else`` clause should not be triggered at all while the break condition occurs.
